src/StreamFactoryInterface.php

```diff
@@ -7,17 +7,32 @@
 interface StreamFactoryInterface
```

Comment:
@mindplay-dk here is what I am referring to:

> Typically, an instance will wrap a PHP stream;

See also `StreamInterface::detach` and `StreamInterface::seek`.

Despite the fact that PSR-7 talks about "streams" in relation to the storage for `StreamInterface` it clearly means resources. Things like callback streams are an interesting case but PSR-7 does nothing to imply interface level support for them.

Based on this evidence, and the fact that all current PSR-7 implements include a resource-backed stream, I think it is perfectly acceptable to define the **default** stream factory as supporting resource-based streams only.